### DISCUSSION TITLE
Continuous Twitch Chat Notice if on Scene A or B

### DISCUSSION THREAD

**Discussion by b3ck** [WarmUpTill/SceneSwitcher#1167] [Q&A]
if I wanted to setup a macro so that if on scene A or B for X amount of minutes (and every X amount of minutes) send chat message in twitch chat how would I do that?

I'm just getting stuck on the logic of things.. can't wrap my head around it, like I know I have to do it with a timer and/or wait action.. but I don't know if I can do it in just one macro or have two other macros that send the message and a timer.. so yeah just confusing myself at this point =/

**Comment by b3ck** [comment]
I'll give you a scenario, if an IRL Streamer puts the stream in a BRB or PRIVACY scene and forgets, it checks if OBS is still on either of the scenes every X amount of minutes and if so it sends a chat message in Twitch chat letting the streamer know they are still in BRB or PRIVACY.

**Comment by WarmUpTill** [comment]
I would set it up with two macros:

1. A macro which handles the sending of the twitch chat minutes every X minutes:

![image](https://github.com/user-attachments/assets/220b762e-2ab7-47e4-9f47-faa88ac03222)

2. A macro which check for the BRB scene and adjust the timer's state accordingly.
Stopping it when not on the BRB scene and restarting it when on the BRB scene.

![image](https://github.com/user-attachments/assets/80190e25-0c32-4fb3-b4e4-fcd843abe71c)

You can [import](https://github.com/WarmUpTill/SceneSwitcher/wiki/Exporting-and-importing-individual-macros) the example using this code:
```
AAANYnic7Vbdb9owEP9XkB8nQAmlTOVt/Zhaad26QreHqZpccwQLx45shxUh/vfdOcSE0lVbH1at2kMk+77vd+e7rFjOhTWODb+tmOY5sCE7vj5uiRmIOWuzgpcOaVOuHNDNcqVARYLRJzOuM5TwtsS7m8vi7B7EJz3y3Poo57wp3gkvjXYX04/Gnxq9tZpZUxbxNjFiPgLvpc4wqhWzkEnnwUaBGXfXpT4uvTe69ou0K4p0j4ph+NJ94HeNoGcymyn8/MX0xOiJDGGNUSNK2Nr+GO4xCYb+aix26MEnckq9z7upaMgVO06qiDZCW/8tbqFFYXd3NN5TQE+phIi7bN2uClmBHGVGhZIewbsyLtxDmQsqd3rUW7c3536/v77dGDhDc5WRX+v2+lvd9G1CunWVzo2fw9I1OgYxqIiofxuhapK8yTIFVw/JWxCCZwdZDto3W0MYpXjhYBLdoYmTEpst3624aNLY5bKlwhFBk6jLnABNhVImk4INE2zC0nJyfWkmciqp+VbMyxy+Y1DOWy61D3IOKMoQzIIr6qB4SLrI98uCjmvKW1YqC7AuoJmukRxcj0CBqCBe1Srt5musAlzX9tKEgKXaoAbrvhkRu/uGhfzHluuqYGNuM/CBGZFoeEeIufgr+BJyFqMDze8UWaueZ+g36ugZ960cnOMZPoGNbF2BZ0Jb5zYOUqFdX3OCoZoQn+7rr2gYOlIXpf/CraQgKOdbqvPm3Twaw/919krXWa+3XUl0/pN1lh4lUfffX2Z1q7/AMour66mnfviYqfTBU68N9J49LCzkmBiCy4YHSdBwfAHXW2qzppPGBAA3M/6F9+UP6cVsf7zyzU/CYZIQAmYOtP+/cpvfFGOpFKPJZIHnY+kVzcAzjV2Mw49u6JV7yIxdUqjkp5PGn4xO9eLsHOwpOGFlUXli1e0OWhWTrChZnHN3Coovt1PuiYIPfm+4a21KLYAwvaymdkxgM8XZrtSJUcYGQwKHtiYo4/BvYLIO/IZN/KFi++vy8XWC1EWTUNfgcwllTYlpsLTbG3T7nYNBJ5seTvpcDPCn7SfuqKRj
```

**Reply by b3ck** [reply]
Awesome, thank you so much, I'm about to go fishing so I gotta hit the road but I will try this when I get back <3
